### ISSUE TITLE
branch_table.lock_key column maybe too short for large tx.

### ISSUE THREAD

**Issue by tq02ksu** [seata/seata#1650]
- [yes] I have searched the [issues](https://github.com/seata/seata/issues) of this repository and believe that this is not a duplicate.

### Ⅰ. Issue Description
I use seata in my business, for a large transaction, the seata server process failure due to can bind a LONG value only for insert into a LONG column.
i was wonder that the lock_key column can be cut off. Suppose not, the lock key is duplicated in lock_table?
### Ⅱ. Describe what happened
I want to cut off the lock key to 128 chars. or remove the field to increase some performance.
  If there is an exception, please attach the exception trace:

> Caused by: java.sql.SQLException: ORA-01461: can bind a LONG value only for insert into a LONG column
>
>        at oracle.jdbc.driver.T4CTTIoer11.processError(T4CTTIoer11.java:494)
>        at oracle.jdbc.driver.T4CTTIoer11.processError(T4CTTIoer11.java:446)
>        at oracle.jdbc.driver.T4C8Oall.processError(T4C8Oall.java:1054)
>        at oracle.jdbc.driver.T4CTTIfun.receive(T4CTTIfun.java:623)
>        at oracle.jdbc.driver.T4CTTIfun.doRPC(T4CTTIfun.java:252)
>        at oracle.jdbc.driver.T4C8Oall.doOALL(T4C8Oall.java:612)
>        at oracle.jdbc.driver.T4CPreparedStatement.doOall8(T4CPreparedStatement.java:226)
>        at oracle.jdbc.driver.T4CPreparedStatement.doOall8(T4CPreparedStatement.java:59)
>        at >oracle.jdbc.driver.T4CPreparedStatement.executeForRows(T4CPreparedStatement.java:910)
>        at oracle.jdbc.driver.OracleStatement.doExecuteWithTimeout(OracleStatement.java:1119)
>        at >oracle.jdbc.driver.OraclePreparedStatement.executeInternal(OraclePreparedStatement.java:3780)
>        at oracle.jdbc.driver.T4CPreparedStatement.executeInternal(T4CPreparedStatement.java:1343)
>        at oracle.jdbc.driver.OraclePreparedStatement.executeLargeUpdate(OraclePreparedStatement.java:3865)
>        at oracle.jdbc.driver.OraclePreparedStatement.executeUpdate(OraclePreparedStatement.java:3845)
>        at oracle.jdbc.driver.OraclePreparedStatementWrapper.executeUpdate(OraclePreparedStatementWrapper.java:1061)
>        at org.apache.commons.dbcp.DelegatingPreparedStatement.executeUpdate(DelegatingPreparedStatement.java:105)
>        at org.apache.commons.dbcp.DelegatingPreparedStatement.executeUpdate(DelegatingPreparedStatement.java:105)
>        at io.seata.core.store.db.LogStoreDataBaseDAO.insertBranchTransactionDO(LogStoreDataBaseDAO.java:387)

> 2019-09-12 14:13:36.546 ERROR[ServerHandlerThread_44_500]io.seata.core.exception.AbstractExceptionHandler.exceptionHandleTemplate:120 -Catch TransactionException while do RPC, request: xid=192.168.1.2:1000:2021840774,branchType=AT,resourceId=jdbc:oracle:thin:@192.168.1.1:1521/ora11g,lockKey=<many table and key, 7k+ chars>

```
Just paste your stack trace here!
```

### Ⅳ. How to reproduce it (as minimally and precisely as possible)

1. simple modify the sample code in /seata/seata-samples/multiple-datasource-mybatis-plus/src/main/java/io/seata/samples/mutiple/mybatisplus/service/impl/OrderServiceImpl.java

```
 package io.seata.samples.mutiple.mybatisplus.service.impl;

import com.baomidou.mybatisplus.extension.service.impl.ServiceImpl;
import io.seata.core.context.RootContext;
import io.seata.samples.mutiple.mybatisplus.common.OperationResponse;
import io.seata.samples.mutiple.mybatisplus.common.order.Order;
import io.seata.samples.mutiple.mybatisplus.common.order.OrderStatus;
import io.seata.samples.mutiple.mybatisplus.common.order.PlaceOrderRequestVO;
import io.seata.samples.mutiple.mybatisplus.config.DataSourceKey;
import io.seata.samples.mutiple.mybatisplus.config.DynamicDataSourceContextHolder;
import io.seata.samples.mutiple.mybatisplus.dao.OrderDao;
import io.seata.samples.mutiple.mybatisplus.service.OrderService;
import io.seata.samples.mutiple.mybatisplus.service.PayService;
import io.seata.samples.mutiple.mybatisplus.service.StorageService;
import io.seata.spring.annotation.GlobalTransactional;
import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import javax.annotation.PostConstruct;
import java.math.BigDecimal;
import java.util.List;
import java.util.stream.Collectors;
import java.util.stream.Stream;


/**
 * @author HelloWoodes
 */
@Service
@Slf4j
public class OrderServiceImpl  extends ServiceImpl<OrderDao, Order> implements OrderService {

    @Autowired
    private OrderDao orderDao;

    @Autowired
    private PayService payService;

    @Autowired
    private StorageService storageService;

    @GlobalTransactional
    @Override
    public OperationResponse placeOrder(PlaceOrderRequestVO placeOrderRequestVO) throws Exception {
        log.info("=============ORDER=================");
        DynamicDataSourceContextHolder.setDataSourceKey(DataSourceKey.ORDER);
        log.info("当前 XID: {}", RootContext.getXID());

        Integer amount = 1;
        Integer price = placeOrderRequestVO.getPrice();

        List<Order> orders = Stream.iterate(1, i -> i + 1).limit(price).map(i -> Order.builder()
                .userId(placeOrderRequestVO.getUserId())
                .productId(placeOrderRequestVO.getProductId())
                .status(OrderStatus.INIT)
                .payAmount(price)
                .build()).collect(Collectors.toList());

        saveBatch(orders);

            log.info("保存订单{}", "成功" );
        // 扣减库存
        boolean operationStorageResult = storageService.reduceStock(placeOrderRequestVO.getProductId(), amount);

        // 扣减余额
        boolean operationBalanceResult = payService.reduceBalance(placeOrderRequestVO.getUserId(), price);

        log.info("=============ORDER=================");
        DynamicDataSourceContextHolder.setDataSourceKey(DataSourceKey.ORDER);

        for (Order order : orders) {
            order.setStatus(OrderStatus.SUCCESS);
            int updateOrderRecord = orderDao.updateById(order);
            log.info("更新订单:{} {}", order.getId(), updateOrderRecord > 0 ? "成功" : "失败");
        }

        return OperationResponse.builder()
                                .success(operationStorageResult && operationBalanceResult)
                                .build();
    }
}

```
2. change the account balance to 5000 (seata_pay.account)

3. issue the request
```
curl -X POST   http://127.0.0.1:8081/order/placeOrder   -H 'Content-Type: application/json'   -d '{
    "userId": 1,
    "productId": 1,
    "price": 1000
}'
```

### Ⅵ. Environment:

- JDK version : jdk8
- OS : windows
- Others: seata develop branch with this patch
https://github.com/seata/seata/pull/1628

**Comment by tq02ksu** [comment]
I checked the source code, if the branch_table.lock_key set no be "empty", the branch lock will not be released after the global transaction committed. It is a big problem. Anyone have some idea about use seata process complicated business is distributed environment? not use db store? or enlarge the column to text? I think it is not a good idea.

**Comment by slievrly** [comment]
How long is the lockkey of a branch transaction in your use?

**Comment by tq02ksu** [comment]
@siliverly, thanks for your concern, in my project, the branch transaction will modify 400 records, 18 chars each id, so total lock_key is about 8k. 
I noticed that the branch_table will not preserve branch log totally. So, i easily change the column to text/clob, it seams to work properly.

**Comment by zjinlei** [comment]
fixed in 1.0.0
